clients/src/main/java/org/apache/kafka/common/requests/OffsetFetchRequest.java

```diff
@@ -38,17 +38,37 @@
     private static final String PARTITION_KEY_NAME = "partition";
 
     public static class Builder extends AbstractRequest.Builder<OffsetFetchRequest> {
+        private static final List<TopicPartition> ALL_TOPIC_PARTITIONS = null;
         private final String groupId;
         private final List<TopicPartition> partitions;
+        private short minVersion = (short) 0;
 
         public Builder(String groupId, List<TopicPartition> partitions) {
             super(ApiKeys.OFFSET_FETCH);
             this.groupId = groupId;
             this.partitions = partitions;
         }
 
+        public static Builder allTopicPartitions(String groupId) {
+            return new Builder(groupId, ALL_TOPIC_PARTITIONS);
+        }
+
+        public boolean isAllTopicPartitions() {
+            return this.partitions == ALL_TOPIC_PARTITIONS;
+        }
+
+        /**
+         * Set the minimum version which we will produce for this request.
+         */
+        public Builder setMinVersion(short minVersion) {
+            this.minVersion = minVersion;
+            return this;
+        }
+
         @Override
         public OffsetFetchRequest build() {
+            if (isAllTopicPartitions() && version() < minVersion)
```

Comment:
Sure, I thought about it too, but thought to keep it in sync with `ListOffsetRequest`. I'll update.